test/libsolidity/smtCheckerTests/loops/for_loop_trivial_condition_3.sol

```diff
@@ -16,3 +16,5 @@ contract C {
 // ----
 // Warning: (115-121): Unused local variable.
 // Warning: (356-370): Assertion violation happens here
+// Warning: (285-290): Underflow (resulting value less than 0) happens here
```

Comment:
Which expression triggered the overflow check here?

Reply:
These are caused by `x = 2` (overflow is checked at assignments) since all knowledge regarding variables that are touched inside the loop are erase.

Reply:
But how can `x=2` cause an underflow?

Reply:
I'm assuming that since all information regarding `x` is erased it chooses any value for the expression, including negative ones. I'll check further.

Reply:
Yes, the reason is that even though `x = 2` is added to the solver at that point, all the information created inside the loop body is popped from the solver after. So in the end when underflow/overflow are actually checked, this information is not there. What happens then is that the expression is a variable without constraints and the solver can give it values that cause underflow/overflow. This is all caused by assignments also triggering underflow/overflow checks. We could also remove that and always check on the right hand side only.

Reply:
I'm just a little confused because assignment involving the same types should never cause an *erflow.